commands/analyze.go

```diff
@@ -39,109 +39,142 @@ func init() {
-			res.Log.Errorf(errStr)
+		if !compatible {
+			errStr := fmt.Sprintf("Error: %s was imported with an incompatible version of RITA", toRunDirty[i].Name())
+			res.Log.Error(errStr)
 			fmt.Println(errStr)
 			continue
 		}
-		toRun = append(toRun, possDB)
+		toRun = append(toRun, toRunDirty[i])
+	}
+
+	//If theres no databases to analyze after filtering everything out
+	//exit
+	if len(toRun) == 0 {
+		return cli.NewExitError("", 255)
 	}
 
 	startAll := time.Now()
 
 	fmt.Println("[+] Analyzing:")
-	for _, db := range toRun {
-		fmt.Println("\t[-] " + db)
+	for i := range toRun {
+		fmt.Println("\t[-] " + toRun[i].Name())
+	}
+
+	var dbNames []string
+	for i := range toRun {
+		dbNames = append(dbNames, toRun[i].Name())
 	}
 	res.Log.WithFields(log.Fields{
-		"databases":  toRun,
+		"databases":  dbNames,
 		"start_time": startAll.Format(util.TimeFormat),
 	}).Info("Preparing to analyze ")
 
-	for _, td := range toRun {
+	for _, ritaDB := range toRun {
 		startIndiv := time.Now()
 		res.Log.WithFields(log.Fields{
-			"database":   td,
+			"database":   ritaDB.Name(),
 			"start_time": startIndiv.Format(util.TimeFormat),
 		}).Info("Analyzing")
-		fmt.Println("[+] Analyzing " + td)
-		res.DB.SelectDB(td)
+		fmt.Println("[+] Analyzing " + ritaDB.Name())
+		res.DB.SelectDB(ritaDB.Name())
 
 		sanitization.SanitizeData(res)
 
-		logAnalysisFunc("Unique Connections", td, res,
+		logAnalysisFunc("Unique Connections", ritaDB.Name(), res,
 			structure.BuildUniqueConnectionsCollection,
 		)
-		logAnalysisFunc("Unique Hosts", td, res,
+		logAnalysisFunc("Unique Hosts", ritaDB.Name(), res,
 			func(innerRes *resources.Resources) {
 				structure.BuildHostsCollection(innerRes)
 				structure.BuildIPv4Collection(innerRes)
 				structure.BuildIPv6Collection(innerRes)
 			},
 		)
-		logAnalysisFunc("Unique Hostnames", td, res,
+		logAnalysisFunc("Unique Hostnames", ritaDB.Name(), res,
 			dns.BuildHostnamesCollection,
 		)
-		logAnalysisFunc("Exploded DNS", td, res,
+		logAnalysisFunc("Exploded DNS", ritaDB.Name(), res,
 			dns.BuildExplodedDNSCollection,
 		)
-		logAnalysisFunc("URL Length", td, res,
+		logAnalysisFunc("URL Length", ritaDB.Name(), res,
 			urls.BuildUrlsCollection,
 		)
-		logAnalysisFunc("User Agent", td, res,
+		logAnalysisFunc("User Agent", ritaDB.Name(), res,
 			useragent.BuildUserAgentCollection,
 		)
-		logAnalysisFunc("Blacklisted", td, res,
+		logAnalysisFunc("Blacklisted", ritaDB.Name(), res,
 			blacklist.BuildBlacklistedCollections,
 		)
-		logAnalysisFunc("Beaconing", td, res,
+		logAnalysisFunc("Beaconing", ritaDB.Name(), res,
 			beacon.BuildBeaconCollection,
 		)
-		logAnalysisFunc("Scanning", td, res,
+		logAnalysisFunc("Scanning", ritaDB.Name(), res,
 			scanning.BuildScanningCollection,
 		)
-		logAnalysisFunc("Cross Reference", td, res,
+		logAnalysisFunc("Cross Reference", ritaDB.Name(), res,
 			crossref.BuildXRefCollection,
 		)
 
-		res.MetaDB.MarkDBAnalyzed(td, true)
```

Comment:
We use to swallow an error in MarkDBAnalyzed which may fail since it interacts with MongoDB